plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TrinoRow.java

```diff
@@ -0,0 +1,129 @@
+import io.trino.spi.type.TinyintType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarbinaryType;
+import io.trino.spi.type.VarcharType;
+import org.apache.iceberg.StructLike;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.plugin.iceberg.util.Timestamps.getTimestampTz;
+import static io.trino.plugin.iceberg.util.Timestamps.timestampTzToMicros;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.type.Decimals.readBigDecimal;
+import static io.trino.spi.type.TimeType.TIME_MICROS;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MICROS;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MICROS;
+import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_MICROSECOND;
+import static java.lang.Float.intBitsToFloat;
+import static java.lang.Math.toIntExact;
+import static java.util.Objects.requireNonNull;
+
+public class TrinoRow
+        implements StructLike
+{
+    private final List<Type> types;
+    private final Block[] blocks;
+    private final int position;
+
+    public TrinoRow(List<Type> types, Block[] blocks, int position)
+    {
+        this.types = requireNonNull(types, "types list is null");
+        this.blocks = requireNonNull(blocks, "blocks array is null");
+        checkArgument(position >= 0, "page position must be positive: %s", position);
+        this.position = position;
+    }
+
+    public int getPosition()
+    {
+        return position;
+    }
+
+    @Override
+    public int size()
+    {
+        return blocks.length;
+    }
+
+    @Override
+    public <T> T get(int i, Class<T> aClass)
+    {
+        Block block = blocks[i].getLoadedBlock();
+        Type type = types.get(i);
+        T value;
+        // TODO: can refactor with IcebergPageSink.getIcebergValue
+        if (block.isNull(position)) {
+            value = null;
+        }
+        else if (type instanceof BigintType) {
+            value = aClass.cast(type.getLong(block, position));
+        }
+        else if (type instanceof IntegerType || type instanceof SmallintType || type instanceof TinyintType || type instanceof DateType) {
+            value = aClass.cast(toIntExact(type.getLong(block, position)));
+        }
+        else if (type instanceof BooleanType) {
+            value = aClass.cast(type.getBoolean(block, position));
+        }
+        else if (type instanceof DecimalType) {
+            value = aClass.cast(readBigDecimal((DecimalType) type, block, position));
+        }
+        else if (type instanceof RealType) {
+            value = aClass.cast(intBitsToFloat(toIntExact(type.getLong(block, position))));
+        }
+        else if (type instanceof DoubleType) {
+            value = aClass.cast(type.getDouble(block, position));
+        }
+        else if (type.equals(TIME_MICROS)) {
+            value = aClass.cast(type.getLong(block, position) / PICOSECONDS_PER_MICROSECOND);
```

Comment:
This logic could ideally be in a shared function, doing mapping reverse to `io.trino.plugin.iceberg.IcebergTypes#convertIcebergValueToTrino`